src/Http/Response.php

```diff
@@ -925,6 +925,12 @@ public function getStatusCode()
      */
     public function withStatus($code, $reasonPhrase = '')
     {
+        if ($code === null) {
+            return $this->_status;
+        }
+        if (!isset($this->_statusCodes[$code])) {
+            throw new InvalidArgumentException('Unknown status code');
```

Comment:
Why is this necessary? It seems wrong the code that you can set any custom code, in the case not found it will just not set any reason phrase.

Reply:
I added it because the implementation of statusCode() explicitly checks if the provided code is known to Cake, and I image there are people relying on always having a valid response code being returned. To me, it doesn't matter if a reason phrase can be added or not, just that the responsecode is valid. But, since Cake already has this list of codes, it seemed like a good start.

Reply:
but what if an application uses custom codes? will that be a Problem then?

Reply:
It will be, just as it was before withStatus() was available. I have no problem with applications using their own statuscodes, but since CakePHP explicitly links to the RFC's for status codes and has its own internal list of valid codes and reason phrases, I'd assume it uses them as well.

I'm not against having people chose their own status code, but it should be a valid code; be that anyting between 100 and 599 or the list in Cake itself doesn't matter to me.